iceoryx_dust/container/include/iox/detail/fixed_position_container.inl

```diff
@@ -51,6 +51,118 @@ inline FixedPositionContainer<T, CAPACITY>::~FixedPositionContainer() noexcept
     }
 }
 
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(const FixedPositionContainer& rhs) noexcept
+{
+    *this = rhs;
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(FixedPositionContainer&& rhs) noexcept
+{
+    *this = std::move(rhs);
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>&
+FixedPositionContainer<T, CAPACITY>::operator=(const FixedPositionContainer& rhs) noexcept
+{
+    if (this != &rhs)
+    {
+        IndexType i = 0U;
+        auto lhsSize = size();
+        auto minSize = algorithm::minVal(lhsSize, rhs.size());
+
+        // copy using copy assignment
+        for (; i < minSize; ++i)
+        {
+            m_data[i] = rhs.m_data[i];
+            m_status[i] = rhs.m_status[i];
+            m_next[i] = rhs.m_next[i];
+        }
+
+        // with rhs.size bigger than this.size: copy further elements from rhs to this
+        for (; i < rhs.size(); ++i)
+        {
+            m_data[i] = rhs.m_data[i];
+            m_status[i] = rhs.m_status[i];
+            m_next[i] = rhs.m_next[i];
+        }
+
+        // with rhs.size smaller than this.size: delete remaining elements of this (erase)
+        for (; i < lhsSize; ++i)
+        {
+            erase(i);
```

Comment:
This too and it will even terminate when the slot behind `i` has the free status